c/example04.c

```diff
@@ -0,0 +1,148 @@
+
+struct Tag {
+    char *resource_id;
+    char *key;
+    char *value;
+    CURL *curl;
+    struct curl_slist *headers;
+};
+
+
+/* Function to initialize the headers */
+void init_headers(struct curl_slist **headers)
+{
+    *headers = curl_slist_append(*headers, "Content-Type: application/json");
+}
+
+/* Function to create a tag */
+struct Tag *create_tag(struct Tag *tag)
+{
+    CURLcode res;
+    char json_data[BUFSIZ];
+    snprintf(json_data, sizeof(json_data), "{\"ResourceIds\": [\"%s\"], \"Tags\": [{\"Key\": \"%s\", \"Value\": \"%s\"}]}", tag->resource_id, tag->key, tag->value);
+
+    curl_easy_setopt(tag->curl, CURLOPT_URL, "https://api.eu-west-2.outscale.com/api/v1/CreateTags");
+
+    /* Set the Content-Type header */
+    curl_easy_setopt(tag->curl, CURLOPT_HTTPHEADER, tag->headers);
+
+    /* Empty post field to indicate we want to send a POST request */
+    curl_easy_setopt(tag->curl, CURLOPT_POSTFIELDS, json_data);
+
+    /* Perform the POST request */
+    res = curl_easy_perform(tag->curl);
+
+    if (res != CURLE_OK) {
+        fprintf(stderr, "Failed to create tag: %s\n", curl_easy_strerror(res));
+    } else {
+        printf("\nTag created successfully:\n");
+        printf("Resource ID: %s\n", tag->resource_id);
+        printf("Key: %s\n", tag->key);
+        printf("Value: %s\n", tag->value);
+    }
+
+    return tag;
+}
+
+/* Function to delete a tag */
+int delete_tag(struct Tag *tag)
+{
+    CURLcode res;
+    char json_data[BUFSIZ];
+    snprintf(json_data, sizeof(json_data), "{\"ResourceIds\": [\"%s\"], \"Tags\": [{\"Key\": \"%s\", \"Value\": \"%s\"}]}", tag->resource_id, tag->key, tag->value);
+
+    /* Setting the URL for delete */
+    curl_easy_setopt(tag->curl, CURLOPT_URL, "https://api.eu-west-2.outscale.com/api/v1/DeleteTags");
+
+    /* Set the Content-Type header */
+    curl_easy_setopt(tag->curl, CURLOPT_HTTPHEADER, tag->headers);
+
+    /* Empty post field to indicate we want to send a POST request */
+    curl_easy_setopt(tag->curl, CURLOPT_POSTFIELDS, json_data);
+
+    /* Perform the POST request */
+    res = curl_easy_perform(tag->curl);
+
+    if (res != CURLE_OK) {
+        fprintf(stderr, "Failed to delete tag: %s\n", curl_easy_strerror(res));
+    } else {
+        printf("\nTag deleted successfully:\n");
+        printf("Resource ID: %s\n", tag->resource_id);
+        printf("Key: %s\n", tag->key);
+        printf("Value: %s\n", tag->value);
+    }
+
+    return 0;
+}
+
+void delete_tag_cleanup(struct Tag **newTag)
+{
+    if (*newTag)
+        delete_tag(*newTag);
+}
+
+int main(int argc, char *argv[])
+{
+    /* Getting the access key / secret key from the environment */
+    const char *ak = getenv("OSC_ACCESS_KEY");
+    const char *sk = getenv("OSC_SECRET_KEY");
+
+    /* Error handling */
+    if (ak == NULL || sk == NULL) {
+        fprintf(stderr, "OSC_ACCESS_KEY or OSC_SECRET_KEY environment variable not set\n");
+        return 1;
```

Comment:
I personally prefer `1/0` to `EXIT_SUCCESS/EXIT_FAILURE` because there's shorter.
But TBH there are a few OS out there that don't use `0/1` for error (But I can't think of any), but `EXIT_SUCCESS/EXIT_FAILURE` is still good practice in theory. 
Anyway, please use either `EXIT_SUCCESS/EXIT_FAILURE` or `0/1`, but don't mix them.